doc/rm/register_tool/index.md

```diff
@@ -571,6 +571,148 @@ There is no way to guarantee that hardware is doing the right thing, but at leas
+
+  A fault attack on the reset line can reset the staged, shadow and committed registers altogether and not generate an error.
+  It may remain undetected by the system.
+  It is thus **highly recommended** to use non-permissive reset values and use software for configuring more permissive values.
+
+- Error hook-ups:
+  - Storage errors signaled to the hardware in `err_storage` are fatal, and should be hooked up directly into the [alert system]({{< relref "hw/ip/alert_handler/doc" >}}).
+  - Update errors signaled to the hardware in `err_update` may be software errors and do not affect the current committed value; thus they can be considered informational bugs that are collected into error status or interrupts, and optionally alerts.
+    They do not generate error responses on the TL-UL system bus interface.
+
+- Hardware and software cost:
+
+  Shadow registers are costly as they imply considerable overheads in terms of circuit area as well as software complexity and performance.
+  They should not be used lightly nor extensively.
+  For example, if a register is reconfigured frequently, using restrictive reset values, possibly combined with parity bits and/or with verification of the register value after writing, might be more suitable.
+  In contrast, for registers that are more static the performance overhead of shadow registers is negligible.
+  Also, for some registers it is not possible to define restrictive reset values, e.g., due to process variation.
+  The use of shadow registers is more justified in such cases.
+  Example use cases for shadow registers are:
+  - Main control registers of cryptographic accelerators such as AES and HMAC where fault attacks could enforce potentially less secure operating modes.
+  - Critical and security-related infrastructure such as:
+    - Alert escalation and sensor configuration registers, and
+    - Countermeasure tuning and functional/bandgap calibration registers.
+
+### Future enhancements
+
+The following features are currently not implemented but might be added in the future.
+
+- Lightweight implementations:
+
+  Depending on the use case and security requirements, lightweight shadow register implementations can be more suitable.
+  Possible options include:
+  - Storing only a hash value or parity bits in the staged and/or shadow register instead of the full data word.
+    This can help to reduce circuit area overheads but may weaken security.
+  - Removing the staged register and relying on software to verify the written value.
+    This helps to reduce circuit area but increases software complexity.
+    If lower security can be tolerated, avoiding the read back after a write can make this implementation transparent to software.
+  - A combination of the two options.
+    For example, removing the staged register and storing parity bits only in the shadow register may be suitable for physical memory protection (PMP) registers.
+
+- Ones' complement inversion in software:
```

Comment:
is it `one's` complement or `ones'` complement? opinions appear to be divided. I'd say leave it as `ones'`, makes a bit more sense (and that's what wikipedia uses so it must be right)

Reply:
I think `ones'` is correct.  I nerded out and found this stackoverflow [discussion](https://stackoverflow.com/questions/2604296/twos-complement-why-the-name-two) - the comment that was most helpful to me was:
> The link "Method of Complements" in this comment provides some explanation. I believe you can rationalize it this way: The "two" in "two's complement" refers to the radix (which is 2 in binary). The "one" in "ones' complement" refers to the fact how you negate a number in this system: For each digit (there are usually multiple or many, hence the plural form), the new digit is computed by subtracting it from 1. By the way, the same would apply for systems that store 10-ary numbers (i.e. decimal) - there it would be "ten's complement" and "nines' complement" – Jo So May 18 at 16:59

Reply:
Never thought that much about it but doing a quick web search I also came to the conclusion to replace one's by ones' complement :-)